pkg/cmd/roachtest/tests/rebalance_load.go

```diff
@@ -160,14 +164,20 @@ func registerRebalanceLoad(r registry.Registry) {
 				case <-time.After(statSamplePeriod):
 				}
 
+				now := timeutil.Now()
 				clusterStoresCPU, err := storeCPUFn(ctx)
 				if err != nil {
 					t.L().Printf("unable to get the cluster stores CPU %s\n", err.Error())
+					continue
 				}
-
-				done, reason = isLoadEvenlyDistributed(clusterStoresCPU, meanCPUTolerance)
+				var curIsBalanced bool
+				curIsBalanced, reason = isLoadEvenlyDistributed(clusterStoresCPU, meanCPUTolerance)
 				t.L().Printf("cpu %s", reason)
-				if done {
+				if !prevIsBalanced && curIsBalanced {
+					balancedStartTime = now
+				}
+				prevIsBalanced = curIsBalanced
```

Comment:
Shouldn't this only change from false to true (never back again?). Otherwise, if it is temporarily unbalanced, this will reset the balancedStartTime the next time it becomes balanced, although maybe that is what is intended here.